File: Runtime/python/Phrost/GeomType.py
import enum

# Imports from your existing adapter file
# (Assuming it's saved as phrost_adapter.py)
try:
    from Phrost import Events
except ImportError:
    print(
        "Could not import Events. Please ensure phrost_adapter.py is in the same directory."
    )


class GeomType(enum.IntEnum):
    """Helper enum to define Geometry types, mapping to their Event ID."""

    POINT = Events.GEOM_ADD_POINT.value
    LINE = Events.GEOM_ADD_LINE.value
    RECT = Events.GEOM_ADD_RECT.value
    FILL_RECT = Events.GEOM_ADD_FILL_RECT.value

File: Runtime/javascript/Phrost/GeomType.js
/**
 * Helper enum to define Geometry types, mapping to their Event ID.
 * * @readonly
 * @enum {number}
 */
const GeomType = Object.freeze({
  POINT: 50, // Events.GEOM_ADD_POINT
  LINE: 51, // Events.GEOM_ADD_LINE
  RECT: 52, // Events.GEOM_ADD_RECT
  FILL_RECT: 53, // Events.GEOM_ADD_FILL_RECT
});

module.exports = GeomType;
